### ISSUE TITLE
Changes to navigation modes

### ISSUE THREAD

**Issue by peastman** [ArtOfIllusion/ArtOfIllusion#64]
This is to continue the discussion that was started at #62.  I think the new navigation modes could use some changes.  Let me go through some of the issues I encountered when working with them.

In Fly and Drive modes, there are cue graphics that appear every time you move the mouse, then disappear after a few seconds.  I see several issues with this:

- It's really distracting having the overlay constantly appearing and disappearing!  It gets in the way of trying to do other things.
- The constant redraws hurt performance.
- It's only relevant to a very specific function (moving the camera along the z axis), so having it appear when you're doing other things isn't useful.
- It's not obvious what the graphics mean.  For example, in Fly mode you get inner and outer circles, connected by a line passing through the mouse position, and a third smaller circle around the mouse position.  Apparently the inner circle is to indicate the central dead zone, but what about the rest of them?
- It's not even clear to me that the graphics are useful.  Users should be able to just think of it as, "Use the scroll wheel to zoom in on whatever the cursor is pointing at."  And mostly, that seems to work pretty well.  Perhaps having a central dead zone makes the behavior feel a little more natural, but there's no need to draw a circle around it.

Another question is whether we can make it more intuitive to the user what these modes mean.  When I first saw a menu with the word "Tray" above every view, I didn't know what it meant.  It wasn't obvious that had anything to do with navigation.

When you go into Fly or Drive mode, it disables the parallel/perspective menu and forces it to perspective mode.  Why?  Is there a reason these modes can't be used in parallel view?

In any case, there's a bug in the implementation of forcing it to perspective mode.  Try the following:

1. Start out looking at a front view, parallel, tray mode.
2. Switch to fly mode.  It disables the parallel/perspective menu and shows a perspective view.
3. Use the scroll wheel to zoom in and out a few times at different places.
4. Switch back to tray mode.  It reenables the menu and claims to be in parallel mode, front view, but the  view is obviously not correct.  For example, I can see the sides of an axis aligned box!
5. Reselecting either parallel mode or front view doesn't fix it.  But if I first select a different view direction (back view, for example) then select front again, that does fix it.

**Comment by Sailsman63** [comment]
A user [on sourceforge](https://sourceforge.net/p/aoi/discussion/47782/thread/2c1a2f05/?limit=25#5b81) recently tried to run AOI under Java 9. In addition to issues with the wrapper `.exe`s,

We've found out that there are issues with the view controls under java 9 on Windows.
As in, they are totally broken. The system seems to treat all mouse buttons equally, and the modifier keys are ignored. Some control is possible through using the old 'Camera Control' View tool buttons, But they do not really work fully.

It is not clear to me if this is a new bug in the JVM, or if they _fixed_ or optimized something, and _we_ were relying on a bug. Possibly related to the changes that caused [This JDK issue](https://bugs.openjdk.java.net/browse/JDK-8187126?jql=text%20~%20%22middle%20click%22%20ORDER%20BY%20created%20DESC)

@peastman Could you try to run AOI under Java 9 on a Mac, so we can see if this is specific to Windows? My linux VM seems to handle some things better, but still not quite right. I'm not sure if that is intrinsic to the OS, or is a weird pass-through of mouse state from the windows host.

**Comment by peteihis** [comment]
I mised/forgot this discussion for a while. Aplogies for that, I'll try to sketch a plan forward.

> It's really distracting having the overlay constantly appearing and disappearing! It gets in the way of trying to do other things.

I'd have to agree. At the cery least a redesign would be needed, but I think some kind of a indicator, what to expect from the scroll action would be useful, ore the way I see it "opligatory" for the two modes.

> "Use the scroll wheel to zoom in on whatever the cursor is pointing at."

That is basically the standard way in most of the rest of the world. _(The AoI-way has been to just zoom in/out to the cneter of the view.)_ However, the the zooming with the mouse cursor as the center can be a bit inpredictable and sometimes there is a user selection wheter to use it of not and even when it is used it only moves the camera along the view plane. In this case the camera turns, so the point that you are aiming at does not stay in one place, which I thought would increase the difficulty in handling it. 

I also wanted to provide the possibility of traveling directly forward/back just by positioning the mouse. Using the center are for that seem like anautar thing. Then there was the decision to be made how steep the turn should be when the mouse cursos is not in teh center area. That made me think of gradually changing the action of linear travel to truning on the spot, whwn the mouse is moving away from the center.

> The constant redraws hurt performance.

So I notice. Unfortunately some form of contant redraw is something we'll have to go into sooner or later. 

> there's a bug -- Aha! took me a few tries to get it.

So there is! What happens is, that the scrolling does not check if the camera has turned from it's original orientation, so it still thinks it is "Front", "Left" ...  if you started in those orientatons. Then switching from Front to Front logically does nothig.

I have found a few other glitches myself. I'll start a PR, to fix those and this one soon.

> Another question is whether we can make it more intuitive to the user what these modes mean.

We should. Usually there are icons for the selection, but on the plain AoI-UI there aren't any on the menubars of the views, so it never crossed my mind to add some and somehow I though, that the place to put icons into would be the DisplayModeIcons plugin, where the "infrastructure" for icons is already existing.
